homeassistant/components/zha/sensor.py

```diff
@@ -236,6 +236,20 @@ async def _refresh(self, time):
             )
 
 
+# pylint: disable-next=hass-invalid-inheritance # needs fixing
+class EnumSensor(Sensor):
+    """Sensor with value from enum."""
+
+    _attr_device_class: SensorDeviceClass = SensorDeviceClass.ENUM
+    _enum: type[enum.Enum] | None = None
+
+    def formatter(self, value: int) -> str | None:
+        """Use name of enum."""
+        if self._enum is None:
```

Comment:
if we have an enum sensor this case should be impossible... maybe we should make this `assert self._enum`?

Reply:
That is indeed better than not checking at all, which is what I did in downstream branches.